module/metrics.go

```diff
@@ -424,3 +424,29 @@ type PingMetrics interface {
 	// NodeInfo tracks the software version, sealed height and hotstuff view of a node
 	NodeInfo(node *flow.Identity, nodeInfo string, version string, sealedHeight uint64, hotstuffCurView uint64)
 }
+
+type HeroCacheMetrics interface {
+	// BucketAvailableSlots keeps track of number of available slots in buckets of cache.
+	BucketAvailableSlots(uint64, uint64)
+
+	// OnSuccessfulWrite is called whenever a new entity is successfully added to the cache.
+	OnSuccessfulWrite()
+
+	// OnEntityEjectedAtFullCapacity is called whenever adding a new entity to the cache results in ejection of another entity.
+	// This normally happens when the cache is full.
```

Comment:
s~(This normally happens) (when the cache is full)~$1 -- and is expected -- $2~ ? :-)

Reply:
https://github.com/onflow/flow-go/pull/2093/commits/ee25f8d0ea77ce4c4a014170bfdb71209409646b